runtime/near-vm-logic/src/logic.rs

```diff
@@ -2376,7 +2376,11 @@ impl<'a> VMLogic<'a> {
         let evicted_ptr = self.ext.storage_get(&key, StorageGetMode::Trie)?;
         let evicted =
             Self::deref_value(&mut self.gas_counter, storage_write_evicted_byte, evicted_ptr)?;
-        let nodes_delta = self.ext.get_trie_nodes_count() - nodes_before;
+        let nodes_delta = self
```

Comment:
In #8386 I changed these to

```
        let nodes_delta;
        {
            #![allow(clippy::arithmetic_side_effects)]
            // FIXME: write a comment explaining why this cannot panic.
            nodes_delta = self.ext.get_trie_nodes_count() - nodes_before;
        };
```

I’m not particularly comfortable replacing these with `checked_sub` by myself – I have no way to tell whether this subtraction always succeeds, and unfortunately returning an `IntegerOverflow` here might end up exposing and accidentally stabilizing internal implementation details to the protocol level, which seems potentially pretty bad.

I think ideally these are left alone for the time being until we can get some dedicated time to review these and validate their behaviour.

Reply:
In my understanding current behaviour in production is to panic when overflow occurs. Using `checked_sub` here changes the behaviour from panicking to returning `Err`, which seems like a safe change to me.

Reply:
This is a _safe_ change in isolation of this pull request. However consider that these errors make it into the execution result of a contract. In that perspective, _once_ we land this PR, we’re effectively stabilizing this behaviour on an overflow, so the question isn’t all about how safe this change is, but whether we’re comfortable with this being a protocol-facing behaviour in the first place.

Admittedly, bringing the whole network down on overflow is not a great behaviour either, but I think this is a good opportunity to think about whether `saturating_sub` wouldn’t be the most appropriate choice here.

Reply:
Alas, the call here is not on me alone – I think it would be ideal to involve people working on the storage stack in reviewing this particular portion of the code, which I think we’d want to have a self-contained PR highlighting just the affected portions of the code for.